### DISCUSSION TITLE
Weather widget layout issues (with vertical taskbar)

### DISCUSSION THREAD

**Discussion by Caspid** [valinet/ExplorerPatcher#2674] [Q&A]
With the W10 taskbar on the side, the weather widget gets cut off, even when "display on 2 lines" is checked.

![Screenshot 2024-01-10 073104](https://github.com/valinet/ExplorerPatcher/assets/2336773/34adf524-68d9-4c8f-98a8-722257197385)

In W10, the behavior was to display the description below the icon:
![Screenshot 2024-01-10 073230_th](https://github.com/valinet/ExplorerPatcher/assets/2336773/653b3c81-6d24-4893-9999-355c1eb455ed)

Does anyone have a fix?

**Comment by pyrates999** [comment]
Does making the taskbar wider on the side help?

**Reply by Caspid** [reply]
Of course it fits if the taskbar is widened, but I don't want it to be wider. I would prefer the text to be on the next line as it was in W10, please.

**Reply by pyrates999** [reply]
How did windows 10 handle it?

**Reply by Caspid** [reply]
There's a screenshot in my original post. There's a line break after the icon.

**Reply by pyrates999** [reply]
Since Microsoft doesn't intend you to put the windows 11 taskbar on the side, that is why it has this behavior.

**Reply by Caspid** [reply]
I know, but that's for the W11 taskbar. This is occurring for the W10 taskbar that's afforded by Explorer Patcher.

**Reply by pyrates999** [reply]
In windows 10, it also does this when you enable weather in EP.  But if weather is disabled in EP, and I enable news and interests with icon and text ticked, it doesn't do this.

Here's how to add weather in windows 11: https://www.microsoft.com/en-us/windows/learning-center/how-to-check-the-weather-in-windows-11

**Reply by Caspid** [reply]
I can't find a News and Interests widget in W11 (even on the W11 taskbar)? "Widgets" is enabled in Settings → Personalization → Taskbar.

**Reply by pyrates999** [reply]
https://www.microsoft.com/en-us/windows/learning-center/best-news-apps-for-windows

**Reply by Caspid** [reply]
Hi, I wonder if you're misunderstanding. The W10 taskbar displayed the weather information on a separate line. The Explorer Patcher W10 taskbar does not.

**Reply by pyrates999** [reply]
I don't think windows 11 has news and interests in its taskbar.

**Reply by Caspid** [reply]
Right, but the W10 taskbar did / does, just the layout is messed up. Asking for a bug fix - maybe I should post this in a different forum.

**Reply by pyrates999** [reply]
Try using window 11's weather widget instead.

**Reply by Caspid** [reply]
I want to use the W10 taskbar (via Explorer Patcher), but have the weather show up properly.

**Reply by pyrates999** [reply]
Ok, please wait for a developer to respond then.

When you enable the windows 10 taskbar, you can try right clicking on it and selecting news and interests as that will show the weather:

![image](https://github.com/valinet/ExplorerPatcher/assets/106135298/d8f45720-bf85-476c-9c8a-1f7a9d165622)

**Reply by pyrates999** [reply]
Just tried to get it working when the taskbar is on the side.  And you can see the description part is actually on 2 lines.

There just isn't enough room for the description.

If you look at the windows 10 weather feature, it only shows the icon and temperature.  There is no description.  So try changing the setting **Layout** to **Icon and Temperature**.

It is able to show the description when the taskbar is on the bottom because the area where it is shown automatically expands to accommodate the text.  When the taskbar is on the side, you need to expand the taskbar until it fits as it behaves differently when it is on the side.

**Reply by Caspid** [reply]
I'm asking for a line break after the icon, so the text (temperature) shows up below,  as it did in W10:

![Screenshot 2024-01-17 161735](https://github.com/valinet/ExplorerPatcher/assets/2336773/d42d6a9a-6989-47ee-a207-9e6eb1e2bf82)


...as opposed to appearing on the same line in W11 (with the W10 taskbar via EP):

![Screenshot 2024-01-17 161444](https://github.com/valinet/ExplorerPatcher/assets/2336773/5adcd4ee-40a4-413a-8cc7-8575d9d39c73)

**Reply by pyrates999** [reply]
Ok, please wait for a developer to respond.
